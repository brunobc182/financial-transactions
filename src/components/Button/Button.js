import React from 'react';
import {
  bool, func, node, string,
} from 'prop-types';
import styled from 'styled-components';
import colors from '../../theme/colors';

const Button = ({
  children, onClick, type, isDebit,
}) => (
  <ButtonComponent type={type} onClick={onClick} isDebit={isDebit}>
    {children}
  </ButtonComponent>
);

const ButtonComponent = styled.button`
  width: 100%;
  max-width: 140px;
  height: 40px;
  box-shadow: 0 6px 10px 0 rgba(0, 0, 0, 0.1);
  background: ${({ isDebit }) => (isDebit ? colors.radicalRed : colors.shamrock)};
  border-radius: 4px;
  outline: none;
  border: none;
  cursor: pointer;
  color: ${colors.white};
  font-size: 14px;
  font-weight: bold;
  padding: 6px 12px;
  text-transform: uppercase;

  :hover {
    background: ${({ isDebit }) => (isDebit ? colors.cinnabar : colors.emerald)};
  }

  :active {
    position: relative;
    top: 1px;
  }
`;

Button.propTypes = {
  children: node,
  isDebit: bool,
  onClick: func,
  type: string,
};

Button.defaultProps = {
  children: 'Button text',
  isDebit: false,
  onClick: null,
  type: 'button',
};

export default Button;
