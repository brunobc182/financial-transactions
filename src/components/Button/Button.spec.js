import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
