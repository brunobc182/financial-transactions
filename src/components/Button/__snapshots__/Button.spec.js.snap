// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="sc-bdVaJa aajhJ"
  onClick={null}
  type="button"
>
  Button text
</button>
`;
